jinahub/segmenters/VADSpeechSegmenter/vad_speech_segmenter.py

```diff
@@ -0,0 +1,144 @@
+__copyright__ = 'Copyright (c) 2020-2021 Jina AI Limited. All rights reserved.'
+__license__ = 'Apache-2.0'
+
+from pathlib import Path
+from typing import Optional, Tuple
+
+import torch
+import torchaudio
+import numpy as np
+import librosa as lr
+from jina import Executor, Document, DocumentArray, requests
+from jina.excepts import BadDocType
+
+
+class VADSpeechSegmenter(Executor):
+    """
+     Segment the speech audio using Silero's Voice Activity Detector (VAD)
+
+    :param normalize: a bool to specify whether to normalize the audio
+     by the sample rate
+    :param dump: a bool to specify whether to dump the segmented audio
+    :param repo: default repo name of silero-vad
+    :param model: default model name of silero-vad
+
+    """
+
+    TARGET_SAMPLE_RATE = 16000
+
+    def __init__(
+        self,
+        normalize: bool = False,
+        dump: bool = False,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        # TODO: remove the following temporary fix for torch hub
+        torch.hub._validate_not_a_forked_repo = lambda *args: True
+        self.model, utils = torch.hub.load(
+            repo_or_dir='snakers4/silero-vad', model='silero_vad', force_reload=True
+        )
+        self.model.eval()  # set model to eval mode
+
+        (_, self.get_speech_ts_adaptive, self.save_audio, *_) = utils
+        self.normalize = self._normalize if normalize else lambda audio, _: audio
+        self.dump = dump
+
+    @requests
+    def segment(
+        self,
+        docs: Optional[DocumentArray] = None,
+        parameters: dict = {},
+        **kwargs,
+    ):
+        """
+        Segment all audio docs to chunks using silero-vad
+        :param docs: documents sent to the segmenter. The docs must have either
+        `blob` or `uri` that ends with either `.mp3` or `.wav`.
+        :param parameters: dictionary to override the default parameters.
+        """
+        if not docs:
+            return
+
+        for doc in docs:
+            if not doc:
```

Comment:
Why is this needed? `DocumentArray` will iterate over `Documents` and they will be valid

Reply:
Right, it's not needed. I have removed it [here](https://github.com/jina-ai/executors/pull/97/commits/d201306b0833330a60d65e770698bbbd768790df).